File: F# Basics/18004.fs
open System


let rec f a b =
    if a <= b then b-a
    else if a%2=1 then 1 + f (a+1) b
    else 1 + f (a/2) b

[<EntryPoint>]
let main _ =
    
    let line = Console.ReadLine().Split ' ' |> Array.map int
    let a = line.[0]
    let b = line.[1]

    printf "%d\n" (f a b)

    0

File: 18004.ml
open Scanf
open Printf
;;
let rec f a b =
  if a <= b then b-a
  else if a mod 2=1 then 1 + f (a+1) b
  else 1 + f (a/2) b
;;
scanf "%d %d\n" (fun a b -> printf "%d\n" (f a b))
;;
